tests/lib/linter/linter.js

```diff
@@ -3408,6 +3425,52 @@ var a = "test2";
             });
         });
 
+        describe("physicalFilenames", () => {
+            it("should allow physicalFilename to be passed on options object", () => {
+                const physicalFilenameChecker = sinon.spy(context => {
+                    assert.strictEqual(context.getPhysicalFilename(), "foo.js");
+                    return {};
+                });
+
+                linter.defineRule("checker", physicalFilenameChecker);
+                linter.verify("foo;", { rules: { checker: "error" } }, { physicalFilename: "foo.js" });
+                assert(physicalFilenameChecker.calledOnce);
+            });
+
+            it("should allow physicalFilename to be passed as third argument", () => {
+                const physicalFilenameChecker = sinon.spy(context => {
+                    assert.strictEqual(context.getPhysicalFilename(), "foo.js");
+                    return {};
+                });
+
+                linter.defineRule("checker", physicalFilenameChecker);
+                linter.verify("foo;", { rules: { checker: "error" } }, { physicalFilename: "foo.js" });
+                assert(physicalFilenameChecker.calledOnce);
+            });
+
+            it("should default physicalFilename to <input> when options object doesn't have filename", () => {
```

Comment:
While I think the suggestion below correctly updates the description to be consistent with the test, the current description raises a good question: if we pass `filename` but not `physicalFilename`, what should `physicalFilename` default to? Right now it would be `filename === "test.js" && physicalFilename === "<input>"` in that case, which seems weird. Should `physicalFilename` default to `filename` instead?

```suggestion
            it("should default physicalFilename to <input> when options object doesn't have physicalFilename", () => {
```

Reply:
Yeah, I think for stdin, filename and physicalFilename should always match.

Reply:
fixed 👍🏻